src/react-components/2d-hud.js

```diff
@@ -37,18 +37,45 @@ TopHUD.propTypes = {
   onToggleSpaceBubble: PropTypes.func
 };
 
-const BottomHUD = ({ onCreateObject }) => (
-  <div className={cx(styles.container, styles.bottom)}>
-    <div
-      className={cx("ui-interactive", styles.iconButton, styles.large, styles.createObject)}
-      title={"Create Object"}
-      onClick={onCreateObject}
-    />
+const mediaPickerInput = "media-picker-input";
+const BottomHUD = ({ onCreateObject, onMediaPicked }) => (
+  <div>
+    {onMediaPicked ? (
```

Comment:
The behavior is similar but not the same. The `onCreateObject` callback passed to `2d-hud` displays the create object dialogue, but tapping the `mediaPickerInput`'s `label`/`icon` skips the dialogue and then calls the `onMediaPicked` callback with the files that were given to the browser by the OS. In the second case we want to call `UIRoot.handleCreateObject` which emits an `"add_media"` event on the scene (aside: is there a reason we emit an event here instead of calling a function?), and so in order to call that from `2d-hud.js` I thought I'd need to pass in a callback function.

Reply:
Ah ok.. It still might read easier if I pass a flag use the `onMediaPicked` callback if the flag is true instead of using the existence of the callback as the flag.